app/experimenter/nimbus-ui/src/components/Summary/EndExperiment/index.tsx

```diff
@@ -0,0 +1,126 @@
+import { ExperimentIdInput } from "../../../types/globalTypes";
+
+const EndExperiment = ({
+  experiment,
+  status,
+}: {
+  experiment: getExperiment_experimentBySlug;
+  status: ReturnType<typeof getStatus>;
+}) => {
+  const [submitError, setSubmitError] = useState<string | null>(null);
+  const [showEndConfirmation, setShowEndConfirmation] = useState(false);
+  const [endRequested, setEndRequested] = useState(false);
+  const isEnding = status.ending || endRequested;
+
+  const [endExperiment, { loading: endExperimentLoading }] = useMutation<
+    { endExperiment: EndExperimentResult },
+    { input: ExperimentIdInput }
+  >(END_EXPERIMENT_MUTATION);
+
+  const toggleShowEndConfirmation = useCallback(
+    () => setShowEndConfirmation(!showEndConfirmation),
+    [showEndConfirmation, setShowEndConfirmation],
+  );
+
+  const onConfirmEndClicked = useCallback(async () => {
+    try {
+      const result = await endExperiment({
+        variables: {
+          input: {
+            id: experiment.id,
+          },
+        },
+      });
+
+      if (
+        !result.data?.endExperiment ||
+        result.data.endExperiment.message !== "success"
+      ) {
+        throw new Error(SUBMIT_ERROR);
+      }
+
+      setEndRequested(true);
+    } catch (error) {
+      setSubmitError(SUBMIT_ERROR);
+    }
+  }, [experiment, endExperiment, setEndRequested]);
+
+  if (submitError) {
+    return (
+      <Alert
+        className="mb-4"
+        variant="warning"
+        data-testid="experiment-end-error"
+      >
+        {submitError}
+      </Alert>
+    );
+  }
+
+  return (
+    <div className="mb-4">
+      {status.live && !isEnding && (
+        <>
+          {showEndConfirmation ? (
+            <Alert variant="warning" data-testid="end-experiment-alert">
+              <p>
+                Are you sure you want to end your experiment? It will turn off
+                the experiment for all users in production.
+              </p>
+
+              <div>
+                <Button
+                  variant="primary"
+                  onClick={onConfirmEndClicked}
+                  disabled={endExperimentLoading}
+                  data-testid="end-experiment-confirm"
+                >
+                  Yes, end the experiment
+                </Button>
+
+                <Button
+                  variant="secondary"
+                  className="ml-2"
+                  onClick={toggleShowEndConfirmation}
+                  disabled={endExperimentLoading}
+                  data-testid="end-experiment-cancel"
+                >
+                  Cancel
+                </Button>
+              </div>
+            </Alert>
+          ) : (
+            <Button
+              variant="primary"
+              onClick={toggleShowEndConfirmation}
+              data-testid="end-experiment-start"
+            >
+              End Experiment
+            </Button>
+          )}
+        </>
+      )}
+
+      {isEnding && (
```

Comment:
I think we should check for `!status.complete && isEnding` here - this message should disappear after the pending end has been approved

Reply:
Oh interesting I assumed `isEndRequested` would return to false after it was moved into a complete state.

Reply:
(I moved the status.live check up to <Summary> btw)